### DISCUSSION TITLE
SBFspot for MacOS (OSX)

### DISCUSSION THREAD

**Discussion by sbf-** [SBFspot/SBFspot#462] [General]
I wonder if someone already tried to compile and run SBFspot on MacOS

**Comment by mincequi** [comment]
Sure ;)

Works on my fork.

**Comment by sbf-** [comment]
Nice... but without bluetooth nor upload daemon?
Having zero experience, what are the steps to take?

**Comment by mincequi** [comment]
True, without Bluetooth nor upload.
From my perspective, the least effort is to port this app to Qt and create a single executable for polling the inverter **and** doing the upload.

I already integrated the reading of an energy meter using Qt and the next steps are:
- port the ethernet stuff to Qt
- using QtService to realize cross plattform service/daemon.
- port bluetooth to Qt
- port PV upload to Qt
- switch to CMake (and drop the other build systems)
- port SQL access

This should be enough for a proper cross-platform solution. Any help is highly appreciated :)

**Comment by wimleers** [comment]
I'm running SBFspot on macOS using https://github.com/nakla/sbfspot 😊

**Reply by sbf-** [reply]
Sure, that´s a possibility too...

**Reply by wimleers** [reply]
But I worry that I'm running too old a version of SBFspot: I'm still on `3.8.4` but per https://github.com/SBFspot/SBFspot/tags, `3.9.7` is the latest…

So I'd actually rather run it natively if I'm otherwise many versions (~1.5 year!) behind. And I'd rather not use the 50-commits-ahead, 156-commits-behind fork by @mincequi at https://github.com/mincequi/SBFspot 😅

I'll give building it on macOS a try at some point. But for now, this will do. I would actually be totally fine continuing to run it through Docker if I could just get it to run the latest version… but for that I'd need to learn more about Docker first 🤓

**Reply by sbf-** [reply]
I guess this belongs to #635

**Reply by wimleers** [reply]
It's definitely associated; I'm mostly looking for the best way to run it long-term.
